### DISCUSSION TITLE
Adobe Animate and Linux

### DISCUSSION THREAD

**Discussion by markkelynack** [airsdk/Adobe-Runtime-Support#2646] [General]
Hi,

I have been asked to get our .air files running on Linux, we use Animate as our main tool.

I have been reading some guides on here on how to install, but these all seem to use a bin folder with the project files in them (Flex I’m guessing)

Question, when publishing from Animate, what would the folder that we copy to Linux need to contain and what would it look like?

**Comment by ajwfrost** [comment]
Hi

For Linux, it doesn't support the ".air" file directly, so you'd have to have the AIR SDK on a Linux machine and then re-package the application using the "bundle" target. That would then create a folder structure that has everything you need to run the application, and you can use normal Linux tools to turn that into an installer/package.

thanks

**Reply by markkelynack** [reply]
Thanks, so this is where i'm a little lost - using Animate I have .swf, app.xml, icon folder as the project files only (plus .p12 cert), is that all I need to package the app?

**Reply by ajwfrost** [reply]
You can use those files too, yes. The key thing is that you'll need to package up the application on a Linux machine... so you'll need to have the AIR SDK on there, and then do:
```
AIRSDK_path/bin/adt -package -storetype pkcs12 -keystore cert.p12 -target bundle MyApp app.xml myapp.swf iconfolder
```
or if you had already created a .air file you should be able to use this:
```
AIRSDK_path/bin/adt -package -storetype pkcs12 -keystore cert.p12 -target bundle MyApp myapp.air
```

It then generates the output in a "MyApp" folder...
